pandas/tests/test_window.py

```diff
@@ -441,6 +441,34 @@ def test_closed(self):
         with pytest.raises(ValueError):
             df.rolling(window=3, closed='neither')
 
+    def test_empty_df_datetime_rolling_sum(self):
```

Comment:
rather than making all separate tests, either use parametrize, or you can consolidate these into 2 tests for rolling (and 2 for expanding)

e.g.

```
@pytest.mark.parametrize('roller', ['1s', 1]):
def tests_empty_df_rolling(self, roller):
    expected = DataFrame()
    result = DataFrame().rolling(roller).sum()
    tm.assert_frame_equal(result, expected)

    expected = DataFrame(index=pd.DatetimeIndex([]))
    result = DataFrame(index=pd.DatetimeIndex([])).rolling(roller).sum()
    tm.assert_frame_equal(result, expected)
```